src/Typography/Typography.component.tsx

```diff
@@ -84,40 +86,52 @@ export interface TypographyProps extends SpaceProps, React.HTMLAttributes<any> {
 
 export interface StyledTypographyProps extends TypographyProps {
     $color?: 'inherit' | 'initial' | string;
+    asVariant?: any;
+    scaleVariant?: any;
 }
 
-export const Typography = styled('span').attrs<TypographyProps>(
-    ({ color }) => ({
-        $color: color,
-        color: undefined,
-        className: 'anchor-typography',
-    })
-)<StyledTypographyProps>`
+export const Typography = styled('span').attrs<TypographyProps>(props => ({
+    $color: props.color,
+    color: undefined,
+    className: 'anchor-typography',
+    asVariant: variant({
+        key: 'typography.as',
+        default: 'none',
+        prop: 'as',
+    })(props),
+    scaleVariant: variant({
+        key: 'typography.scale',
+        default: 'none',
+        prop: 'scale',
+    })(props),
+}))<StyledTypographyProps>`
 
     box-sizing: border-box;
     margin: 0;
 
     // Variant styles
-    ${props =>
+    ${({ asVariant, scaleVariant }) =>
         css({
             fontFamily: 'base',
             fontSize: th('typography.fontSize'),
             lineHeight: th('typography.lineHeight'),
             fontWeight: th('typography.fontWeight'),
             textAlign: 'inherit',
             color: 'inherit',
-            ...variant({
-                key: 'typography.as',
-                default: 'none',
-                prop: 'as',
-            })(props),
-            ...variant({
-                key: 'typography.scale',
-                default: 'none',
-                prop: 'scale',
-            })(props),
+            // Don't include variants that are css-blocks
```

Comment:
I implemented it this way (spreading if it can, and otherwise including it normally afterward) to preserve deduplication of properties when it can (notice that we didn't break any tests, just added new ones).

We can just nix all this and always include the variants the usual way (with no deduplication) if this solution has strayed too far from god.